portfolio/src/main/java/com/google/sps/data/Video.java

```diff
@@ -19,6 +20,7 @@
     private boolean isPublic;
     // Fetch when analyzing.
     private String captions;
+    private VideoException error = null;
     
     public Video(String videoId, String title, String description, String thumbnailUrl, boolean isPublic) 
       throws MalformedURLException{
```

Comment:
Do we need this constructor? Is it ever used?

Reply:
Currently, we're using this in a few places like videoFromHttpRequest and videoFromPlaylistItem. If we wanted, we could probably make constructors that take in http requests or playlist items instead.